### ISSUE TITLE
Form elements go blank after phx-change

### ISSUE THREAD

**Issue by svileng** [phoenixframework/phoenix_live_view#1037]
### Environment

* Elixir version (elixir -v): `1.9.4`
* Phoenix version (mix deps): `1.5.3`
* Phoenix LiveView version (mix deps): `0.13.3`
* NodeJS version (node -v): `12.13.1`
* NPM version (npm -v): `6.12.1`
* Operating system: MacOS Catalina 10.15.5
* Browsers you attempted to reproduce this bug on (the more the merrier): Firefox 77
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: no

### Actual behavior

Link to repo: https://github.com/svileng/phoenix_live_view_bug_demo

1. Run server, go to `localhost:4000/form`
1. Type in the first input, the text gets duplicated in the second input
1. Now click on the second input, and make a change

As soon as you do Step 3 the first input goes blank.

### Expected behavior

The first input should stay as before.

**Comment by josevalim** [comment]
The server is the one that dictates the value of the inputs when they are not focused. Since you are not keeping the value for @foo like you do for @bar, it is indeed disappearing, because the server is effectively sending an empty input. You need to either use changesets, which track all changes for you, or keep the values of the inputs accordingly.

**Comment by svileng** [comment]
@josevalim Thanks for the quick reply. I do use a changeset for the form, but this is still a problem for me because the second field I want to update (e.g. the `bar` in this case) is not an attribute in the changeset/struct.

Is there are a workaround to be able to use a changeset AND update a custom field at the same time?

**Comment by chrismccord** [comment]
You need to explicitly set the value via an assign if you are not using changesets (or another FormData protocol datastructure). So `text_input f, :bar, value: @bar`

**Comment by svileng** [comment]
@chrismccord Thanks for the reply, I'm actually doing exactly this for the `:bar` input, so I'm sure what are you suggesting? 

Also please see my reply to @josevalim, he is suggesting switching over to using changesets (which I normally do) but I'd like to have custom fields which I can manually set the value to, in addition to a changeset for the `form_for` helper. Is this possible at the moment?

**Comment by josevalim** [comment]
You are using a changeset but you are not updating the changeset when the inputs change. So you need to either do that or track @foo the same way you track @bar. I recommend running mix phx.gen.live in a live app to get a grasp how LV works with changesets.

**Comment by bar** [comment]
@josevalim I'm sorry but I don't understand what you mean.

**Comment by svileng** [comment]
@josevalim I tried using a changeset for all values in the form. But then I needed to perform a side-effect when the form changes and set a flag to use in the form. This caused the same issue to happen—fields go blank.

I went back to my little bug demo and changed it to update both `foo` and `bar` when the `foo` field changes. When the `bar` field changes, `foo` will go blank, even though the value I'm assigning to the `socket` has nothing to do with the changeset or the form. The change can be seen here:

https://github.com/svileng/phoenix_live_view_bug_demo/blob/master/lib/liveviewbug_web/live/form_live.ex

Is it by design that you cannot use the `phx_change` callback for anything else than validation?
